src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -342,36 +334,29 @@ static void uart_driver_do_read(Context *ctx, term msg)
 
         term bin = term_create_uninitialized_binary(count, &ctx->heap, glb);
         uint8_t *bin_buf = (uint8_t *) term_binary_data(bin);
+
         uart_read_bytes(uart_data->uart_num, bin_buf, count, portMAX_DELAY);
 
```

Comment:
You need to make sure there is enough memory available to create this tuple using port_ensure_available:

https://github.com/atomvm/AtomVM/blob/1bedd1c454bcd91562859cd301406128c84645ff/src/libAtomVM/port.c#L47

Sorry this function is currently undocumented, but it's use is important.

Reply:
`memory_ensure_free` is already used at the top of the `if` block. Isn't it enough?

Reply:
It should be enough. See my comment below.